include/deal.II/base/parameter_handler.h

```diff
@@ -1393,7 +1399,9 @@ class ParameterHandler : public Subscriptor
    * @endcode
    */
   std::ostream &
-  print_parameters(std::ostream &out, const OutputStyle style) const;
+  print_parameters(std::ostream &    out,
+                   const OutputStyle style,
+                   const bool &      sort_alphabetical = true) const;
```

Comment:
one thing I just realized, is that this is also not intuitive. Imagine me doing:
```
print_parameters(...,...,true);
print_parameters(...,...,false);
```
Notice that print parameters now modifies the state of the object and the second call still keeps things sorted? Not sure how to best deal with this. Maybe it needs to be a class property after all.

Reply:
Yes, the object state is modified after the first call to `print_parameters`.

I don't know how to make it work even if making it a class property! Where would the sort happen? Whenever a parameter is declared?

Reply:
Just as a minor comment on top of all of the other things: We generally don't mark scalar arguments (like `bool`) as references.